models/discordactions.js

```diff
@@ -336,6 +339,90 @@ const updateIdleUsersOnDiscord = async () => {
   };
 };
 
+const updateUsersNicknameStatus = async (lastNicknameUpdate) => {
```

Comment:
improve/ refactor this function on below points 
1. Use a cache: You can cache the results of the usersCurrentStatus and usersFutureStatus queries to avoid having to run them every time the function is called.
2. Use a batch update: You can use a batch update to update the nicknames of multiple users at once. This will improve the performance of the function if there are a large number of users to update.
3. Use a worker pool: You can use a worker pool to update the nicknames of multiple users in parallel. This will further improve the performance of the function, especially if there are a large number of users to update.

Reply:
For 1 & 3, Again, as I've asked above: for I would require time to read and understand about worker pool. Please tell me if it's okay if I can take this as a separate extension and enhancement of this task.

2. This function is already being called separately in batches of 4 for users.